collector/consumer/exporter/otelexporter/instrument.go

```diff
@@ -58,47 +71,140 @@ func (i *instrumentFactory) createNewInstrument(metricName string, kind MetricAg
 	}
 }
 
-func (i *instrumentFactory) recordGaugeAsync(metricName string, singleGauge model.GaugeGroup) {
-	i.gaugeChan.put(&singleGauge, i.logger)
-	if i.isGaugeAsyncInitialized(metricName) {
-		return
+// only support TraceAsMetric and TcpRttMills now
+func (i *instrumentFactory) recordLastValue(metricName string, singleGauge *model.GaugeGroup) error {
+	if item, ok := i.aggregators.Load(metricName); ok {
+		aggregator := item.(*defaultaggregator.DefaultAggregator)
```

Comment:
The `DefaultAggregator` is not designed for use with multiple instances. The best practice is to insert different metrics into a single aggregator to optimize the performance. I know you want to dump different metrics using different observer callbacks, then you better redesign a new aggregator or modify the current implementation.

Reply:
By expanding the gauge into a gauge group, `recordLastValue()` is using a single DefaultAggregator now